src/main/java/de/tum/in/www1/artemis/domain/enumeration/TestCaseVisibility.java

```diff
@@ -0,0 +1,5 @@
+package de.tum.in.www1.artemis.domain.enumeration;
+
+public enum TestCaseVisibility {
```

Comment:
In case we will use this in `ProgrammingExerciseTestCase` and `Feedback`, we should rename this to just Visibility.
Please also add some JavaDoc documentation and explain in detail what the different cases mean